cmd/clusterctl/cmd/generate_provider.go

```diff
@@ -82,7 +82,7 @@ func init() {
 		"Bootstrap provider and version (e.g. kubeadm:v0.3.0)")
 	generateProviderCmd.Flags().StringVarP(&gpo.controlPlaneProvider, "control-plane", "c", "",
 		"ControlPlane provider and version (e.g. kubeadm:v0.3.0)")
-	generateProviderCmd.Flags().StringVar(&gpo.targetNamespace, "target-namespace", "",
+	generateProviderCmd.Flags().StringVar(&gpo.targetNamespace, "target-namespace", "n",
```

Comment:
In order to set flag shorthand you have to use `StringVarP` (otherwise you are changing the default value)
Same for other commands

Reply:
Fixed in 96661ce